components/notebook-controller/controllers/notebook_controller.go

```diff
@@ -215,6 +216,31 @@ func (r *NotebookReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		return ctrl.Result{}, err
 	}
 
+	// Check if annotations for the Notebook instance imply a requirement for notebook restart.
+	// This annotation is supposed to be added once a ConfigMap used in the notebook is added/updated.
+
+	annotations := instance.GetAnnotations()
+	notebookRestart, ok := annotations[AnnotationNotebookRestart]
+
+	if ok && notebookRestart == "true" {
+
+		// find the pod associated with the notebook instance
+		foundPod := &corev1.Pod{}
+		err = r.Get(ctx, types.NamespacedName{Name: instance.Name + "-0", Namespace: instance.Namespace}, foundPod)
+		if err != nil && apierrs.IsNotFound(err) {
+			log.Info(fmt.Sprintf("No Pods are currently running for Notebook Server: %s in namesace: %s.", instance.Name, instance.Namespace))
+		} else if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		// delete the pod associated with the notebook instance so it can be restarted
+		r.Delete(context.TODO(), foundPod)
```

Comment:
For deletion to work, we need to provide delete access to the service account.
Please include `delete` in this list. https://github.com/opendatahub-io/kubeflow/blob/491c2777e65b6b7b323131f02f38aeacf1818c2d/components/notebook-controller/config/rbac/role.yaml#L28